### PR TITLE
Add initial skeleton code

### PR DESCRIPTION
**Pull request by LiliC**
This modifies the https://github.com/pmorie/osb-starter-pack skeleton code based on the habitat-service-broker project.

As discussed this first part only adds the skeleton. Subsequent PRs will add the actual functionality to the `pkg/user/logic.go` functions.

### CODE REVIEW

File: pkg/user/logic.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+func NewBusinessLogic(o Options) (*BusinessLogic, error) {
+	return &BusinessLogic{
+		async: o.Async,
+	}, nil
+}
+
+// BusinessLogic provides an implementation of the broker.BusinessLogic
+// interface.
+type BusinessLogic struct {
+	// Indiciates if the broker should handle the requests asynchronously.
```
Comment (krnowak) [inline pkg/user/logic.go]: Indicates

File: pkg/user/logic.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+import (
+	"sync"
+
+	osb "github.com/pmorie/go-open-service-broker-client/v2"
+	"github.com/pmorie/osb-broker-lib/pkg/broker"
+)
+
+// NewBusinessLogic is a hook that is called with the Options the program is run with.
+func NewBusinessLogic(o Options) (*BusinessLogic, error) {
```
Comment (krnowak) [inline pkg/user/logic.go]: Why passing Options by value?

Comment (LiliC) [reply pkg/user/logic.go]: Not sure I understand you question. :)

Comment (krnowak) [reply pkg/user/logic.go]: Why `o Options` instead of `o *Options`? We are just going to read it, so making a copy is probably unnecessary, no?

Comment (LiliC) [reply pkg/user/logic.go]: No reason I think, can switch.

Comment (krnowak) [reply pkg/user/logic.go]: Cool, then please do and with that change - LFAD.

File: pkg/user/cli.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package user
```
Comment (LiliC) [inline pkg/user/cli.go]: There is only one thing I have a question about... This is called `user` pkg, which made sense in the broker starter pack, but not sure the naming is good here. Any thoughts or suggestions on this @krnowak?

Comment (krnowak) [reply pkg/user/cli.go]: What was the meaning of `user` before?

Comment (LiliC) [reply pkg/user/cli.go]: It had mutliple pkgs before where the apis were defined, server pkg etc. So the user pkg was meant to signal that this is where the custom code each user defines for their own broker.

So in our case maybe `habitat` would make sense? Or just `broker` and then we can leave `logic.go`?

Comment (krnowak) [reply pkg/user/cli.go]: Yeah, I was thinking about `broker` too. Or `obs`, but I think `broker` is better.

Comment (LiliC) [reply pkg/user/cli.go]: `broker` makes more sense, IMO. Will rename to that, okay?

**Comment by LiliC** [comment]
@krnowak I updated the PR to reflect your suggestions and the agreed pkg name change. PTALA. Thanks!

**Comment by LiliC** [comment]
> Are you going to change this "user" package name also in pmorie's osb-starter-pack?

Will make a suggestion to him today. Think it would make sense.

**Pull request merged**